300-CNAB-security.md

```diff
@@ -0,0 +1,163 @@
+- Send the data that has been signed to the CNAB Registry
+
+In regard to the final two steps, specific implementations may require different ordering of operations. Thus, a CNAB client MAY perform these steps in any order that does not sacrifice the integrity of the cryptographic signatures.
+
+### Key Management
+
+A client SHOULD maintain a cache of trusted public keys. This cache MAY be directly managed, and this cache MAY adopt a strategy of Trust On First Use (TOFU), whereby the first time a package is fetched from a trusted source, the client may also request the public keys from that server. Other than the TOFU case, clients SHOULD NOT attempt to fetch release keys and packages from the same source at the same time. Timestamp, distribution, and root keys (all part of the TUF specification) MUST be managed according to the TUF specification.
+
+## Provenance and Software Supply Chain
+
+The previous section of the specification deals with managing the integrity of the process of transferring a CNAB bundle to and from a CNAB Registry.
+
+In addition to verifying integrity, there is a second related set of verifications that may be performed on a CNAB. These are related to _provenance_.
+
+Provenance captures concern tracing the origin of a package and its components, and ascertaining the integrity of the steps in the process of composing the package from its components. Compare this with the idea of the _software supply chain_, which is a process for composing software in such a way that it preserves provenance in a verifiable way.
+
+This section specifies a provenance model for CNAB based on the [In-Toto specification](https://github.com/in-toto/docs/blob/v0.9/in-toto-spec.md). This specification pays less attention to the software supply chain (e.g. the specific set of steps), focusing instead on how this information is to be stored and verified in a CNAB bundle.
+
+In-Toto, upon which this part of the specification is built, describes two major concepts: A _layout_ that describes the steps which must be performed to establish provenance for a bundle, and multiple _links_, which tie a cryptographic attestation (along with some metadata) to a step in the layout. For example, if a layout has three steps, verification of that layout would require three links -- one for each step. Layouts may be linear, but they also may be tree-shaped, where one layout invokes other "sublayouts", which in turn each may invoke further sublayouts.
+
+For CNAB, bundles may have associated layouts, and such layouts may be satisfied by links.
+
+### Attaching Layouts and Links to Bundles
+
+For layouts and links to function properly, they must be stored in proximity to the bundle.
+
+The "default layout" may be stored inside of the `custom` section of the bundle descriptor.
+
+EXAMPLE
```

Comment:
missing an example here but that's cool for a follow up PR if intended

Reply:
Yeah, followup. I need to ask the In Toto devs a few things.